src/compiler/checker.ts

```diff
@@ -13420,41 +13474,85 @@ namespace ts {
-                if (attribProperties.length === 0) {
-                    return "";
-                }
-                // Element Attributes has one property, so the element attributes type will be the type of the corresponding
-                // property of the class instance type
-                else if (attribProperties.length === 1) {
-                    return attribProperties[0].name;
-                }
-                // More than one property on ElementAttributesProperty is an error
+            if (!_jsxElementPropertiesName) {
+                const jsxElementAttribPropInterfaceSym = getPropertiesFromJsxElementAttributesProperty();
+                // JSX.ElementAttributesProperty [type]
+                const jsxElementAttribPropInterfaceType = jsxElementAttribPropInterfaceSym && getDeclaredTypeOfSymbol(jsxElementAttribPropInterfaceSym);
+                // The properties of JSX.ElementAttributesProperty
+                const propertiesOfJsxElementAttribPropInterface = jsxElementAttribPropInterfaceType && getPropertiesOfType(jsxElementAttribPropInterfaceType);
+
+                // if there is a property in JSX.ElementAttributesProperty
+                // i.e.
+                // interface ElementAttributesProperty {
+                //     props: {
+                //         children?: any;
+                //     };
+                // }
+                if (propertiesOfJsxElementAttribPropInterface) {
+                    // Element Attributes has zero properties, so the element attributes type will be the class instance type
+                    if (propertiesOfJsxElementAttribPropInterface.length === 0) {
+                        _jsxElementPropertiesName = "";
+                    }
+                    // Element Attributes has one property, so the element attributes type will be the type of the corresponding
+                    // property of the class instance type
+                    else if (propertiesOfJsxElementAttribPropInterface.length === 1) {
+                        _jsxElementPropertiesName = propertiesOfJsxElementAttribPropInterface[0].name;
+                    }
+                    // More than one property on ElementAttributesProperty is an error
+                    else {
+                        error(jsxElementAttribPropInterfaceSym.declarations[0], Diagnostics.The_global_type_JSX_0_may_not_have_more_than_one_property, JsxNames.ElementAttributesPropertyNameContainer);
+                        _jsxElementPropertiesName = undefined;
+                    }
+                }
                 else {
-                    error(attribsPropTypeSym.declarations[0], Diagnostics.The_global_type_JSX_0_may_not_have_more_than_one_property, JsxNames.ElementAttributesPropertyNameContainer);
-                    return undefined;
+                    // No interface exists, so the element attributes type will be an implicit any
+                    _jsxElementPropertiesName = undefined;

```

Comment:
Yep if we can't find the JSXElement. May be it is better to be empty string to indicate that we already done this